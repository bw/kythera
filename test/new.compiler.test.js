require("jest")
const t = require("./util").testOutput

describe("new", () => {
	t("int", "new int")
	t("float", "new float")
	t("bool", "new bool")
	t("str", "new str")
	t("null", "new null")
	t("type", "new type")
	t("fn", 'new fn<int> null')
	t("fn with fn as parameter", 'new fn<fn<> null> null')
	t("obj", `new obj{int a, str b,}`)
})