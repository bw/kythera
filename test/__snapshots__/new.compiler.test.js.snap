// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new bool 1`] = `"KYTHERA.type.PRIMITIVES[\\"bool\\"].makeNew();"`;

exports[`new float 1`] = `"KYTHERA.type.PRIMITIVES[\\"float\\"].makeNew();"`;

exports[`new fn 1`] = `"new KYTHERA.type(\\"fn\\", { parameters: [KYTHERA.type.PRIMITIVES[\\"int\\"]], returns: KYTHERA.type.PRIMITIVES[\\"null\\"]}).makeNew();"`;

exports[`new fn with fn as parameter 1`] = `"new KYTHERA.type(\\"fn\\", { parameters: [new KYTHERA.type(\\"fn\\", { parameters: [], returns: KYTHERA.type.PRIMITIVES[\\"null\\"]})], returns: KYTHERA.type.PRIMITIVES[\\"null\\"]}).makeNew();"`;

exports[`new int 1`] = `"KYTHERA.type.PRIMITIVES[\\"int\\"].makeNew();"`;

exports[`new null 1`] = `"KYTHERA.type.PRIMITIVES[\\"null\\"].makeNew();"`;

exports[`new obj 1`] = `"new KYTHERA.type(\\"obj\\", {\\"a\\": KYTHERA.type.PRIMITIVES[\\"int\\"],\\"b\\": KYTHERA.type.PRIMITIVES[\\"str\\"],}).makeNew();"`;

exports[`new str 1`] = `"KYTHERA.type.PRIMITIVES[\\"str\\"].makeNew();"`;

exports[`new type 1`] = `"KYTHERA.type.PRIMITIVES[\\"type\\"].makeNew();"`;
